### PR TITLE
WIP: Contrib reactphp

### PR DESCRIPTION
**Pull request by wfleurant**
The table will automatically refresh once a second. It has a blocking call to sqlite-db, but is useful for those wanting to save peerstats / datausage for a graph or summary report. Below is an early screenshot of the index view (before twbs/css):  

peerstats_index

### CODE REVIEW

**Comment by ansuz** [comment]
This has been sitting here for a week. Any objections to merging?

**Comment by ansuz** [comment]
ping @lgierth @kpcyrd

**Comment by ghost** [comment]
let's do it

File: contrib/reactphp/src/Toolshed/Toolshed.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        } else {
+            return "Memory ReactPHP: " . $bytesphp . ", cjdns: " . $bytesalloc;
+        }
+    }
+
+    /* */
+    static function publictoip6($pubkey = false) {
+        if (!$pubkey) { return false; }
+        $bin = '../../publictoip6 ';
+        $h   = popen($bin . $pubkey, 'r');
```
Comment (kpcyrd) [inline contrib/reactphp/src/Toolshed/Toolshed.php:65]: Assuming `$pubkey` can be controlled by a user, we have an issue here (didn't verify)

**Comment by kpcyrd** [comment]
I think it's really interesting to have a practical example for the cjdns admin api, due to it's size we could consider moving this in it's own repository :)

**Comment by wfleurant** [comment]
thanks again kp, so the only input i found is originating from the cjdns admin. 

the other user input for hits a sqlite library which is definitely escaped. the last is any phluid server.php  request->get('user-input') responsibility. it _should*_ render text for templates and for console. i'll verify this and also serve it up for fuzz testing. 

if it goes in a separate repo, it'll rot. May be if we get another head on it.. **/me nudges**  @dansup

**Comment by wfleurant** [comment]
fuzzing causin a ruckus.. 

```
PHP Fatal error:  Uncaught exception 'InvalidArgumentException' 
with message 'Invalid request string' in 
/.../cjdns/contrib/reactphp/vendor/guzzlehttp/psr7/src/functions.php:445
```

still crashes with latest guzzlehttp/psr7 (dev-master 5882861)

**Comment by wfleurant** [comment]
and yea.. other breakage.. 

```
PHP Fatal error:  Uncaught exception 'InvalidArgumentException' 
with message 'Invalid request string' in 
/.../cjdns/contrib/reactphp/vendor/guzzlehttp/psr7/src/functions.php:469
```

not even looking further..

File: contrib/reactphp/views/layout.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    <thead class="smallfont">
+        <tr>
+            <th data-field="state"     data-sortable="true" class="col-md-0">
+                Connectivity
+            </th>
+            <th data-field="ipv6"      data-sortable="true" class="col-md-0">
+                cjdns Address
+            </th>
+            <th data-field="bytesIn"   data-sortable="true" class="col-md-0">
```
Comment (wfleurant) [inline contrib/reactphp/views/layout.php]: we once had this code to make it pretty
```
/* 75lb/usage-stats */
	function lbbytes (bytes){
		var kilobyte = 1024,
			megabyte = kilobyte * 1024,
			gigabyte = megabyte * 1024,
			terabyte = gigabyte * 1024;
		if ((bytes >= 0) && (bytes < kilobyte)) {
			return bytes + " B";
		} else if ((bytes >= kilobyte) && (bytes < megabyte)) {
			return (bytes / kilobyte).toFixed(2) + " KB";
		} else if ((bytes >= megabyte) && (bytes < gigabyte)) {
			return (bytes / megabyte).toFixed(2) + " MB";
		} else if ((bytes >= gigabyte) && (bytes < terabyte)) {
			return (bytes / gigabyte).toFixed(2) + " GB";
		} else if (bytes >= terabyte) {
			return (bytes / terabyte).toFixed(2) + " TB";
		} else {
			return bytes + " B";
		}
	}

```

**Pull request closed without merging**